File: 2401. Longest Nice Subarray/Solution2401.java
public class Solution2401 {
    public int longestNiceSubarray(int[] nums) {
        int maxLen = 0, n = nums.length;
        int left = 0, curMask = 0;
        for (int right = 0; right < n; right++) {
            while ((curMask & nums[right]) > 0) {
                curMask ^= nums[left++];
            }
            curMask |= nums[right];
            maxLen = Math.max(maxLen, right - left + 1);
        }
        return maxLen;
    }    
}

// https://leetcode.cn/problems/longest-nice-subarray/solution/bao-li-mei-ju-pythonjavacgo-by-endlessch-z6t6/

File: 2401. Longest Nice Subarray/Solution2401.py
from typing import List


class Solution2401:
    def longestNiceSubarray(self, nums: List[int]) -> int:
        left = mask = maxLen = 0
        for right, cur in enumerate(nums):
            while (mask & nums[right]) != 0:
                mask ^= nums[left]
                left += 1
            mask |= nums[right]
            maxLen = max(maxLen, right - left + 1)
        return maxLen

# https://leetcode.cn/problems/longest-nice-subarray/solution/bao-li-mei-ju-pythonjavacgo-by-endlessch-z6t6/